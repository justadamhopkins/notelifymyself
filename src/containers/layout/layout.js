import React, { useState } from 'react'
import { Header } from '../../components/navigation/header'

const Layout = () => {
  const [sideDrawVisible, setDrawState] = useState(false)


  const sideDrawToggleHandler = () => {
    setDrawState(!sideDrawVisible)
  }

  return (
    <React.Fragment>
      <Header
        sideDrawToggle={sideDrawToggleHandler}
        isVisible={sideDrawVisible}
      />
      <main />
    </React.Fragment>
  )
}

export default Layout
