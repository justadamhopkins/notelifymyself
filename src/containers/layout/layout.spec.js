import React from 'react'
import { mount } from 'enzyme'
import { Layout } from './layout'


describe('<Layout/>', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })
  const props = {
    addCategory: jest.fn(),
    category: { 111: { category: 'Nature', _id: 111 } }
  }
  it('Should render component', () => {
    const layout = mount(<Layout {...props}/>)
    expect(layout).toMatchSnapshot()
  })
  it('initializes the sidedrawvisible state', () => {
    const layout = mount(<Layout {...props}/>)
    expect(layout.find('.Hamburger').length).toEqual(1)
    expect(layout.find('.animated').exists()).toBe(false)
  })
  it('sideDrawToggleHandler toggles the sidedraw/top nav state', () => {
    const layout = mount(<Layout {...props}/>)
    layout.find('.Hamburger').simulate('click')
    layout.update()
    expect(layout.find('.animated').exists()).toBe(true)
    expect(layout.find('.open').length).toBe(2)
  })
  it('should add category', () => {
    const layout = mount(<Layout {...props}/>)
    layout.find('.category').simulate('submit')
    expect(props.addCategory).toHaveBeenCalledTimes(1)
  })
})
