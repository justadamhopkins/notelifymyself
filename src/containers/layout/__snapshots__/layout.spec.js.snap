// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> Should render component 1`] = `
ReactWrapper {
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "getWrappingComponentRenderer": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "isVisible": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": <div
            class="Sidedraw close"
          >
            <ul
              class="NavigationItems"
            >
              <li>
                Test 1
              </li>
              <li>
                Test 1
              </li>
              <li>
                Test 1
              </li>
              <li>
                Test 1
              </li>
              <li>
                Test 1
              </li>
            </ul>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavigationItems />,
            "className": "Sidedraw close",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": Object {
                "instance": <ul
                  class="NavigationItems"
                >
                  <li>
                    Test 1
                  </li>
                  <li>
                    Test 1
                  </li>
                  <li>
                    Test 1
                  </li>
                  <li>
                    Test 1
                  </li>
                  <li>
                    Test 1
                  </li>
                </ul>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <NavigationItem>
                      Test 1
                    </NavigationItem>,
                    <NavigationItem>
                      Test 1
                    </NavigationItem>,
                    <NavigationItem>
                      Test 1
                    </NavigationItem>,
                    <NavigationItem>
                      Test 1
                    </NavigationItem>,
                    <NavigationItem>
                      Test 1
                    </NavigationItem>,
                  ],
                  "className": "NavigationItems",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Test 1",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": <li>
                        Test 1
                      </li>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Test 1",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Test 1",
                      ],
                      "type": "li",
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Test 1",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": <li>
                        Test 1
                      </li>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Test 1",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Test 1",
                      ],
                      "type": "li",
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Test 1",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": <li>
                        Test 1
                      </li>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Test 1",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Test 1",
                      ],
                      "type": "li",
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Test 1",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": <li>
                        Test 1
                      </li>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Test 1",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Test 1",
                      ],
                      "type": "li",
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Test 1",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": <li>
                        Test 1
                      </li>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Test 1",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Test 1",
                      ],
                      "type": "li",
                    },
                    "type": [Function],
                  },
                ],
                "type": "ul",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": undefined,
          "isVisible": false,
          "toggleSideDraw": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": <div
            class="Wrapper close"
          >
            <header
              class="Header"
            >
              <div
                class="Hamburger "
              >
                <div
                  class="bar1"
                />
                <div
                  class="bar2"
                />
                <div
                  class="bar3"
                />
              </div>
            </header>
            <main />
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Header
                isVisible={false}
                sideDrawToggle={[Function]}
              />,
              <main />,
            ],
            "className": "Wrapper close",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isVisible": false,
                "sideDrawToggle": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": <header
                  class="Header"
                >
                  <div
                    class="Hamburger "
                  >
                    <div
                      class="bar1"
                    />
                    <div
                      class="bar2"
                    />
                    <div
                      class="bar3"
                    />
                  </div>
                </header>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Hamburger
                    animateHamBurger={false}
                    toggleSideDraw={[Function]}
                  />,
                  "className": "Header",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "animateHamBurger": false,
                      "toggleSideDraw": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": <div
                        class="Hamburger "
                      >
                        <div
                          class="bar1"
                        />
                        <div
                          class="bar2"
                        />
                        <div
                          class="bar3"
                        />
                      </div>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <div
                            className="bar1"
                          />,
                          <div
                            className="bar2"
                          />,
                          <div
                            className="bar3"
                          />,
                        ],
                        "className": "Hamburger ",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": <div
                            class="bar1"
                          />,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "className": "bar1",
                          },
                          "ref": null,
                          "rendered": Array [
                            undefined,
                          ],
                          "type": "div",
                        },
                        Object {
                          "instance": <div
                            class="bar2"
                          />,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "className": "bar2",
                          },
                          "ref": null,
                          "rendered": Array [
                            undefined,
                          ],
                          "type": "div",
                        },
                        Object {
                          "instance": <div
                            class="bar3"
                          />,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "className": "bar3",
                          },
                          "ref": null,
                          "rendered": Array [
                            undefined,
                          ],
                          "type": "div",
                        },
                      ],
                      "type": "div",
                    },
                    "type": [Function],
                  },
                ],
                "type": "header",
              },
              "type": [Function],
            },
            Object {
              "instance": <main />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "main",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isVisible": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": <div
              class="Sidedraw close"
            >
              <ul
                class="NavigationItems"
              >
                <li>
                  Test 1
                </li>
                <li>
                  Test 1
                </li>
                <li>
                  Test 1
                </li>
                <li>
                  Test 1
                </li>
                <li>
                  Test 1
                </li>
              </ul>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavigationItems />,
              "className": "Sidedraw close",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": Object {
                  "instance": <ul
                    class="NavigationItems"
                  >
                    <li>
                      Test 1
                    </li>
                    <li>
                      Test 1
                    </li>
                    <li>
                      Test 1
                    </li>
                    <li>
                      Test 1
                    </li>
                    <li>
                      Test 1
                    </li>
                  </ul>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <NavigationItem>
                        Test 1
                      </NavigationItem>,
                      <NavigationItem>
                        Test 1
                      </NavigationItem>,
                      <NavigationItem>
                        Test 1
                      </NavigationItem>,
                      <NavigationItem>
                        Test 1
                      </NavigationItem>,
                      <NavigationItem>
                        Test 1
                      </NavigationItem>,
                    ],
                    "className": "NavigationItems",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "Test 1",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": <li>
                          Test 1
                        </li>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Test 1",
                        },
                        "ref": null,
                        "rendered": Array [
                          "Test 1",
                        ],
                        "type": "li",
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "Test 1",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": <li>
                          Test 1
                        </li>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Test 1",
                        },
                        "ref": null,
                        "rendered": Array [
                          "Test 1",
                        ],
                        "type": "li",
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "Test 1",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": <li>
                          Test 1
                        </li>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Test 1",
                        },
                        "ref": null,
                        "rendered": Array [
                          "Test 1",
                        ],
                        "type": "li",
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "Test 1",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": <li>
                          Test 1
                        </li>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Test 1",
                        },
                        "ref": null,
                        "rendered": Array [
                          "Test 1",
                        ],
                        "type": "li",
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "Test 1",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": <li>
                          Test 1
                        </li>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Test 1",
                        },
                        "ref": null,
                        "rendered": Array [
                          "Test 1",
                        ],
                        "type": "li",
                      },
                      "type": [Function],
                    },
                  ],
                  "type": "ul",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "isVisible": false,
            "toggleSideDraw": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": <div
              class="Wrapper close"
            >
              <header
                class="Header"
              >
                <div
                  class="Hamburger "
                >
                  <div
                    class="bar1"
                  />
                  <div
                    class="bar2"
                  />
                  <div
                    class="bar3"
                  />
                </div>
              </header>
              <main />
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Header
                  isVisible={false}
                  sideDrawToggle={[Function]}
                />,
                <main />,
              ],
              "className": "Wrapper close",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "isVisible": false,
                  "sideDrawToggle": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": <header
                    class="Header"
                  >
                    <div
                      class="Hamburger "
                    >
                      <div
                        class="bar1"
                      />
                      <div
                        class="bar2"
                      />
                      <div
                        class="bar3"
                      />
                    </div>
                  </header>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Hamburger
                      animateHamBurger={false}
                      toggleSideDraw={[Function]}
                    />,
                    "className": "Header",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "animateHamBurger": false,
                        "toggleSideDraw": [Function],
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": <div
                          class="Hamburger "
                        >
                          <div
                            class="bar1"
                          />
                          <div
                            class="bar2"
                          />
                          <div
                            class="bar3"
                          />
                        </div>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <div
                              className="bar1"
                            />,
                            <div
                              className="bar2"
                            />,
                            <div
                              className="bar3"
                            />,
                          ],
                          "className": "Hamburger ",
                          "onClick": [Function],
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": <div
                              class="bar1"
                            />,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "className": "bar1",
                            },
                            "ref": null,
                            "rendered": Array [
                              undefined,
                            ],
                            "type": "div",
                          },
                          Object {
                            "instance": <div
                              class="bar2"
                            />,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "className": "bar2",
                            },
                            "ref": null,
                            "rendered": Array [
                              undefined,
                            ],
                            "type": "div",
                          },
                          Object {
                            "instance": <div
                              class="bar3"
                            />,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "className": "bar3",
                            },
                            "ref": null,
                            "rendered": Array [
                              undefined,
                            ],
                            "type": "div",
                          },
                        ],
                        "type": "div",
                      },
                      "type": [Function],
                    },
                  ],
                  "type": "header",
                },
                "type": [Function],
              },
              Object {
                "instance": <main />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "main",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__linkedRoots__): Array [],
  Symbol(enzyme.__unrendered__): <Layout />,
  Symbol(enzyme.__updatedBy__): null,
}
`;
